### PR TITLE
[Feature] Reorganizes parts of process pages

### LINKED ISSUE #9637: ✨ Extra chip on process title
## Description

The header used in an individual process is getting reused on all 3 pages (main page, advertisement info page, and assessment plan page). This is misleading and confusing.

The chip label that we see only refers to the "Advertisement information" page.

## 🦋 Expected Behaviour

Each of these three pages should have it's own header.

1. On main process page: KEEP current title and subtitle and REMOVE chip
2. On advertisement information: CHANGE title and subtitle, KEEP chip label
4. On assessment plan page: REPLACE by MOVING up the title and subtitle currently under the breadcrumbs, and updating the copy in the "organize assessment approach" instructions.

### Advertisement information page
**New title:**
`Advertisement information`

**New subtitle:**
`Define the information and requirements for this recruitment process.`

### Assessment plan
**New title:**
`Assessment plan`

**New subtitle:**
`Select, organize and define the assessments used to evaluate each skill in the advertisement.`

**New copy**

```
Use this section to define which assessments will be used as part of your assessment process. Make sure every skill is assessed at least once to complete your assessment plan.

You can also change the order in which you plan to perform these evaluations. The only exceptions are the “Application screening” and the “Screening questions (at the time of application)” assessments which will always be the first and second steps in any pool advertisement.
```

## 📸 Screenshot

<details><summary>CURRENT process header</summary>
<p>

image

</p>
</details> 

<details><summary>CURRENT advertisement information header</summary>
<p>

image

</p>
</details>  

<details><summary>CURRENT assessment plan</summary>
<p>

image

</p>
</details> 

<details><summary>NEW advertisement information header</summary>
<p>

image

</p>
</details> 

<details><summary>NEW assessment plan header and copy</summary>
<p>

image

</p>
</details> 

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] On main process page: KEEP current title and subtitle and REMOVE chip
- [ ] On advertisement information: CHANGE title and subtitle, KEEP chip label
- [ ] On assessment plan page: REPLACE by MOVING up the title and subtitle currently under the breadcrumbs, and updating the copy in the "organize assessment approach" instructions.
- [ ] `<title>` meta remains unchanged.

### PR DESCRIPTION
**Pull request by mnigh**
🤖 Resolves #9637.

## 👋 Introduction

This PR reorganizes parts of process pages based on the acceptance criteria in the linked issue.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Navigate to process page `/admin/pools/:pool-id`
2. Verify that changes reflect the acceptance criteria in the linked issue in English and French

> [!IMPORTANT]  
> Some of the copy from this linked issue was changed by https://github.com/GCTC-NTGC/gc-digital-talent/issues/10195. I've done my best to incorporate it.

## 📸 Screenshots
Screen Shot 2024-05-01 at 14 37 51
Screen Shot 2024-05-01 at 14 38 03
Screen Shot 2024-05-01 at 15 19 29

### CODE REVIEW

File: apps/web/src/pages/Pools/PoolLayout.tsx
Comment (yonikid15) [inline apps/web/src/pages/Pools/PoolLayout.tsx:92]: Can you change this to a variable? That way it is just a reference later on and we don;t need to execute the function every time we need to check it.

```tsx
  let heroTitle: string | undefined = getShortPoolTitleLabel(intl, pool);
  if (currentPage?.link.url.includes("edit")) {
    heroTitle = currentPage?.title;
  }
  if (currentPage?.link.url.includes("plan")) {
    heroTitle = currentPage?.title;
  }

  let heroSubtitle: string | undefined = subTitle;
  if (currentPage?.link.url.includes("edit")) {
    heroSubtitle = currentPage?.subtitle;
  }
  if (currentPage?.link.url.includes("plan")) {
    heroSubtitle = currentPage?.subtitle;
  }
```

File: apps/web/src/pages/Pools/PoolLayout.tsx
Comment (yonikid15) [inline apps/web/src/pages/Pools/PoolLayout.tsx:134]: We need to update this to use the validation for the assessment plan builder badge when on said page. 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/9637-advertisement-assessment-pages/apps/web/src/utils/poolUtils.tsx#L420-L424

```tsx
export const getPoolCompletenessBadge = (completeness: PoolCompleteness) => {
  const badgeInfo = poolCompletenessMap.get(completeness);


  return badgeInfo ?? defaultCompleteness;
};
```

**Review by yonikid15 — CHANGES_REQUESTED** [review]
Looks good so far 👏, just couple things I noted.

Also, we need to fix/update the AssessmentPlanBuilder page badge as well. Either the validation is incorrect or it's using the different badge. Might be out of scope of this issue though.

image

image

File: apps/web/src/pages/Pools/PoolLayout.tsx
Comment (mnigh) [reply apps/web/src/pages/Pools/PoolLayout.tsx]: See: 05e481b764b3add25120500807cc5aef8c8ab8c4

File: apps/web/src/pages/Pools/PoolLayout.tsx
Comment (mnigh) [reply apps/web/src/pages/Pools/PoolLayout.tsx]: I actually followed an anti-pattern from the beginning defining a component within another component (@esizer has created an [issue](https://github.com/GCTC-NTGC/gc-digital-talent/issues/10228) for this for existing code). I've refactored it so that the functions are outside the component, please see: ca5d06ebd9f2dc4ad7e2cbb4e086922938b39031

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/OrganizeSection.tsx
Comment (mnigh) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/OrganizeSection.tsx:213]: You are right, we are using straight quotes moving forward. This copy is not new, I didn't change it.

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/OrganizeSection.tsx
Comment (mnigh) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/OrganizeSection.tsx:204]: See my note from this PR's description:

> [!IMPORTANT]  
> Some of the copy from this linked issue was changed by https://github.com/GCTC-NTGC/gc-digital-talent/issues/10195. I've done my best to incorporate it.

@Jerryescandon since you were the designer for both that issue and this one, could you have a look to confirm this is what is expected since they overlap?

Comment (Jerryescandon) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/OrganizeSection.tsx:204]: Yup, that's correct, the copy ticket ended up being worked un before this one but Matt's assumption is correct.
"...Make sure every essential skill is assessed" is the accurate copy moving forward.

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/OrganizeSection.tsx
Comment (Jerryescandon) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/OrganizeSection.tsx:213]: > You are right, we are using straight quotes moving forward. This copy is not new, I didn't change it.

We are!? 😅 haha I didnt know that. Is that an A11y consideration? I'll try to keep it in mind for all new stuff.

**Comment by mnigh** [comment]
> Also, we need to fix/update the AssessmentPlanBuilder page badge as well. Either the validation is incorrect or it's using the different badge. Might be out of scope of this issue though.

I think you are right, this is an existing bug. The badge is using `getPoolCompletenessBadge` but it should be a `getAssessmentPlanCompletenessBadge` which is not a function that even exists 😟.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/d846fe55ee3cffe07ab41f31bd1df36e6a5056ef/apps/web/src/pages/Pools/AssessmentPlanBuilderPage/AssessmentPlanBuilderPage.tsx#L100-L101

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/OrganizeSection.tsx
Comment (mnigh) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/OrganizeSection.tsx:213]: Not a11y-related. The decision was made for code simplicity.

Comment (yonikid15) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/OrganizeSection.tsx:225]: Ah okay, I guess well need a new issue to remove all instances of it. I can make one.

File: apps/web/src/pages/Pools/PoolLayout.tsx
Comment (yonikid15) [reply apps/web/src/pages/Pools/PoolLayout.tsx]: Looks good!

File: apps/web/src/pages/Pools/PoolLayout.tsx
Comment (yonikid15) [reply apps/web/src/pages/Pools/PoolLayout.tsx]: I'm still seeing an issue. However, it's not on the assessment plan builder page, that seems to working correctly, so you got that right. We just need to transfer what you did to ViewPoolPage.tsx as well, which might just be adding the missing fields. However, that's getting handled on [Vachan's PR](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10237) 👍 so this should be good to go.

https://github.com/GCTC-NTGC/gc-digital-talent/assets/22059495/88bc01dd-3ee7-4b1e-affd-c83b4005b124

**Review by yonikid15 — APPROVED** [review]
Looks greeeeat 🐯

File: apps/web/src/pages/Pools/PoolLayout.tsx
Comment (mnigh) [reply apps/web/src/pages/Pools/PoolLayout.tsx]: @yonikid15 maybe wait until #10237 is merged into `main` and then merge `main` into this branch to verify it is actually fixed?

Comment (yonikid15) [reply apps/web/src/pages/Pools/PoolLayout.tsx]: Ya we can wait 👍

Comment (mnigh) [reply apps/web/src/pages/Pools/PoolLayout.tsx]: @yonikid15 merged and ready for re-review: edaa39a4fc9bf5c457418e664dfe0bf76808919b.

**Review by yonikid15 — APPROVED** [review]
Approved x2

**Pull request merged**
